src/s390x/zipl.rs

```diff
@@ -29,22 +39,159 @@ pub fn chreipl<P: AsRef<Path>>(dev: P) -> Result<()> {
     Ok(())
 }
 
-/// Runs `zipl` based on Ignition and BLS configuration in `boot`.
-pub fn zipl<P: AsRef<Path>>(boot: P) -> Result<()> {
+fn secure_execution_is_enabled() -> Result<bool> {
+    let sysfs_flag = "/sys/firmware/uv/prot_virt_guest";
+    match File::open(sysfs_flag) {
+        Err(e) if e.kind() == std::io::ErrorKind::NotFound => Ok(false),
+        Err(e) => Err(e).with_context(|| format!("reading {}", sysfs_flag)),
+        Ok(mut f) => {
+            let mut buffer = String::new();
+            f.read_to_string(&mut buffer)?;
+            Ok(buffer.trim() == "1")
+        }
+    }
+}
+
+fn find_files<P: AsRef<Path> + ?Sized>(
+    config_path: &P,
+    f: impl Fn(&PathBuf) -> bool,
+) -> Result<Vec<PathBuf>> {
+    let mut entries = read_dir(&config_path)
+        .with_context(|| format!("reading directory {}", config_path.as_ref().display()))?
+        .filter_map(Result::ok)
+        .filter_map(|e| if f(&e.path()) { Some(e.path()) } else { None })
+        .collect::<Vec<_>>();
+    entries.sort();
+    Ok(entries)
+}
+
+fn regenerate_initrd<P: AsRef<Path>>(source: P, files: &[PathBuf]) -> Result<NamedTempFile> {
```

Comment:
Now that we're just concatenating, we could use the `Initrd` API from `crate::io` to build our mini-initrd. That should simplify this function a lot.

Reply:
Hmm, not sure I follow. We're not actually adding symlinks here, no? But also, the bit you're referring to is in `Initrd::from_reader_filtered`, which we shouldn't have to invoke here since we're not reading in any initrd, just creating a new one.

To clarify, I was thinking something like this:

```rust
let mut initrd = Initrd::default();
for file in files {
    initrd.add(&file, std::fs::read(&file)?);
}
// std::io:copy from initrd.to_bytes() to tempfile
```

We could add an `Initrd` API so that it can directly read from the target files when streaming out to be more efficient, but meh doesn't seem worth it given that the files we're dealing here shouldn't be very large.

Reply:
ahh, ok. now got what you mean! yep, that makes sense